test/exercises/diffusion/heat_2d_dist_buffers.chpl

```diff
@@ -0,0 +1,215 @@
+/*
+  A distributed 2D finite-difference heat/diffusion equation solver
+
+  Computation is executed over a 2D distributed array.
+  The array distribution is managed by the `Block` distribution.
+  Tasks are spawned manually with a `coforall` loop and synchronization
+  is done manually using a `barrier`. Halo regions are shared across
+  locales manually via halo/buffer arrays.
+*/
+
+import BlockDist.Block,
+       Collectives.barrier,
+       Time.stopwatch;
+
+// create a stopwatch to time kernel execution
+var t = new stopwatch();
+config const writeTime = false;
+
+// compile with `-sRunCommDiag=true` to see comm diagnostics
+use CommDiagnostics;
+config param RunCommDiag = false;
+
+// declare configurable constants with default values
+config const nx = 256,      // number of grid points in x
+             ny = 256,      // number of grid points in y
+             nt = 50,       // number of time steps
+             alpha = 0.25,  // diffusion constant
+             solutionStd = 0.222751; // known solution for the default parameters
+
+// define distributed domains and block-distributed array
+const indices = {0..<nx, 0..<ny},
+      indicesInner = indices.expand(-1),
+      Indices = Block.createDomain(indices);
+
+// define distributed 2D arrays over the above domain
+var u, un: [Indices] real = 1.0;
+
+// apply initial conditions
+u = 1.0;
```

Comment:
Note that `u` is double-initialized to 1.0 here due to the initializing expression above.